src/main/java/bowling/Game.java

```diff
@@ -0,0 +1,66 @@
+package bowling;
+
+import bowling.exception.BowlingException;
+import bowling.exception.GameOverException;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class Game {
+    private static int NUMBER_OF_FRAMES = 10;
+    private static int LAST_FRAME_INDEX = NUMBER_OF_FRAMES - 1;
+    private Frame[] gameFrames = new Frame[NUMBER_OF_FRAMES];
+    private int currentFrameIndex = 0;
+    private Set<Frame> applyNextRollBonus = new HashSet<>();
+    private Frame applyNextNextRollBonus;
+    private boolean gameOver = false;
+
+    public Game() {
+        for (int frameNumber = 0; frameNumber < LAST_FRAME_INDEX; frameNumber++) {
+            gameFrames[frameNumber] = new Frame();
+        }
+        gameFrames[LAST_FRAME_INDEX] = new Frame(true);
+    }
+
+    public void roll(int numberOfPinsKnockedDown) throws BowlingException {
+        if (gameOver) {
```

Comment:
There's quite a bit of logic that could be split out into 'sections' or 'phases', or in other words, split out into separate methods.

E.g have a method to handle bonus points, have a method to handle frame index changes etc.